src/box/iproto.cc

```diff
@@ -3460,6 +3573,26 @@ iproto_send_start_msg(void)
 		iproto_do_cfg(&iproto_threads[i], &cfg_msg);
 }
 
+void
+iproto_drop_connections(void)
+{
+	static struct latch latch = LATCH_INITIALIZER(latch);
+	latch_lock(&latch);
+	for (int i = 0; i < iproto_threads_count; i++) {
+		struct iproto_cfg_msg *cfg_msg =
+			(struct iproto_cfg_msg *)xmalloc(sizeof(*cfg_msg));
+		iproto_cfg_msg_create(cfg_msg, IPROTO_CFG_DROP_CONNECTIONS);
+		iproto_do_cfg_async(&iproto_threads[i], cfg_msg);
+	}
+
+	for (int i = 0; i < iproto_threads_count; i++) {
+		while (!iproto_threads[i].tx.drop_finished)
+			fiber_cond_wait(&iproto_threads[i].tx.drop_cond);
+		iproto_threads[i].tx.drop_finished = false;
```

Comment:
It looks like we don't need to have these flag and condition variable per thread.

Reply:
Fixed. Patch is in the one of the previous comments.